lib/ea_restaurant_data_loader_web/controllers/oauth2_controller.ex

```diff
@@ -0,0 +1,19 @@
+defmodule EaRestaurantDataLoaderWeb.Controllers.Oauth2Controller do
+  use EaRestaurantDataLoaderWeb, :controller
+  alias EaRestaurantDataLoader.Lib.Utils.Oauth2Util
+  alias EaRestaurantDataLoader.Lib.Utils.ApplicationUtil
+  alias EaRestaurantDataLoader.Lib.Services.Oauth2Service
+  alias EaRestaurantDataLoader.Lib.ErrorHandlers.BadRequest
+
+  def login(conn, _params) do
+
+    try do
```

Comment:
Try block should only cover the piece with problems. Try this:
```
[client_id, client_secret] =
  try do
    [basic_auth | _] = conn |> get_req_header("authorization")
    [client_id, client_secret] = Oauth2Util.get_client_credentials_from_basic_auth(basic_auth)
  rescue
    _ in MatchError -> raise BadRequest
  end
.
.
.
```

Reply:
Done.